gnocchi/rest/aggregates/processor.py

```diff
@@ -167,16 +169,34 @@ def aggregated(timeseries, aggregation, from_timestamp=None,
                     'timespan (%.2f%%)' % (needed_percent_of_overlap,
                                            percent_of_overlap))
 
-        if aggregation in AGG_MAP:
-            values = AGG_MAP[aggregation](values, axis=1)
-        elif aggregation == 'count':
-            values = numpy.count_nonzero(~numpy.isnan(values), axis=1)
+        try:
+            granularity, times, values, is_aggregated = (
+                agg_operations.evaluate(operations, key, times, values,
+                                        False, references[key]))
+        except agg_operations.InvalidOperations as e:
+            raise UnAggregableTimeseries("Invalid operations: %s" % e.reason)
```

Comment:
Food for thought: do we really need this `UnAggregableTimeseries` here or could we just use ` except InvalidOperations` directly in `api.py`?

Reply:
Honestly, if InvalidOperations is raised, it's a bug too, that mean the generated object built with voluptuous is malformed. I replaced it by a RuntimeError

Reply:
About already present UnAggregableTimeseries I will remove them in an separate patch.